students/MicahBraun/Lesson 03/mailroom.py

```diff
@@ -0,0 +1,227 @@
+                                        print('{:>30s}'.format(data_list[i][0]))        # only print i (not cut_off + i)
+                                        continue
+                                    else:
+                                        print('{:>30s}'.format(data_list[i][0]), '{:>35s}'.format(data_list[cut_off + i][0]))
+                            else:
+                                if i == 0:                                              # list length even
+                                    print(spacing)
+                                    print('{:>20s}'.format(str(data_list[i][0])))       # print Donor Name header
+                                    print(spacing)
+                                else:
+                                    print('{:>15s}'.format(data_list[i][0]), '{:>30s}'.format(data_list[cut_off + i][0]))
+
+                            add_donation_count = 1                                  # add new item to new name index
+                            donation_amt = int(input("Enter in the donation amount from Donor {0}: $".format(full_name)))
+                            print('{0} has donated ${1}'.format(full_name, donation_amt))
+                            get_index = 0                                           # donation index (-1) end of list
+                            for item in range(0, len(data_list)):                   # for items in data_list
+                                if data_list[item][0] == full_name:                 # if item at[0] == full_name
+                                    get_index = item                                # set index to item
+                                    break
+                            data_list[get_index].append(donation_amt)               # append donation amount to end of current lst [don_idx]
+                            data_list[get_index].append(add_donation_count)         # append donation count to end
+                            if data_list[get_index][2] == 1:                        # if donation count is 1 (it should be)
+                                avg = data_list[get_index][1]                       # average variable assignment
+                            data_list[get_index].append(avg)                        # append average to end of current index
+
+                            email_display = str(input("Display Donor Email? (Y/N): "))      # ask user if they want to
+                            if email_display.upper() == "Y":                                # print email message
+                                print(spacing)                                              # if Y, then print
+                                print('Dear {0}, \n\nThank you for your continued support through your \
+contribution of ${1} towards our Foundation\'s fundraising goal.\n\nBest wishes,\n\
+Foundation Board of Directors\n'.format(data_list[get_index][0], data_list[get_index][1]))
+                                print(spacing)
```

Comment:
this function is too complex, too many logic paths. look for ways to simplify the code and separate into smaller functions.